src/pages/pokedex-site.tsx

```diff
@@ -125,28 +150,47 @@ export default function PokedexSite() {
                         />
                     )}
                 </div>
-                <div id="pokemon-card-wrapper" className="col-span-2">
-                    {filteredList.map((pokemon) => {
-                        return (
-                            <PokemonCard
-                                key={pokemon.name}
-                                pokemon={pokemon}
-                                onMouseEnter={() => {
-                                    if (
-                                        pokemon.sprites.front_default === null
-                                    ) {
-                                        return;
-                                    }
+                {dexLayout === "list" && (
+                    <div id="pokemon-card-wrapper" className="col-span-2">
+                        {filteredList.map((pokemon) => {
+                            return (
+                                <PokemonListCard
+                                    key={pokemon.name}
+                                    pokemon={pokemon}
+                                    onMouseEnter={() => {
+                                        if (
+                                            pokemon.sprites.front_default ===
+                                            null
+                                        ) {
+                                            return;
+                                        }
 
-                                    setSpriteImage(
-                                        pokemon.sprites.front_default
-                                    );
-                                }}
-                                onMouseLeave={() => setSpriteImage(undefined)}
-                            />
-                        );
-                    })}
-                </div>
+                                        setSpriteImage(
+                                            pokemon.sprites.front_default
+                                        );
+                                    }}
+                                    onMouseLeave={() =>
+                                        setSpriteImage(undefined)
+                                    }
+                                />
+                            );
+                        })}
+                    </div>
+                )}
+                {dexLayout === "grid" && (
+                    <div id="pokemon-card-wrapper" className="col-span-3">
```

Comment:
There is no need to wrap this in 2 divs. One defining all the prop is enough.

Since both grids (the outer one and the inner one) are 3-column based, it would not even be necessary to create a new one here, although it's an option to explicitely separate them fro each other.

But yeah, al least saving one of the divs would be great due to markup reasons.